### ISSUE TITLE
[Elasticsearch-Connector] Missing support for global filters

### ISSUE THREAD

**Issue by McManning** [elastic/search-ui#758]
Support top level filters (all, none, any). Example of configuration below.

**Describe the bug**

Playing around with the ES connector, I've ran into issues with it not correctly passing along global filters as laid out in the [search-ui docs](https://docs.elastic.co/search-ui/api/core/configuration#filters-global-filters) to searchkit. 

My configuration is more or less as follows:

I've got an index with documents in the form:
```json
{
  "kind" : "Autoclave",
  "name" : "Model 3850 EP Autoclave",
  "description" : "",
  "category" : "Instrument",
  "attribute_facets" : [
    "Access Restriction(s)",
    "Part of Collection"
  ]
}
```

And a connector configuration as:

```tsx
const config: SearchDriverOptions = {
    searchQuery: {
        search_fields: {
            name: {
                weight: 3
            },
            description: {},
            // and so forth
        },
        result_fields: {
            name: {
                raw: {},
                snippet: {
                    size: 100,
                    fallback: true,
                }
            },
            // and so forth
        },
        
        facets: {
            "category.keyword": {
                type: 'value',
                size: 100
            },
            "kind.keyword": {
                type: 'value',
                size: 100
            },
        },

        filters: [
            {
                field: 'provider.id.keyword',
                type: 'any',
                values: [
                    '00000174-b680-e5d9-b8fb-15ae80000000', // Stone Lab
                    '0000014d-91eb-0b07-8ac7-287f80000000' // Genomics Shared Resource
                ]
            },
            {
                field: 'kind.keyword',
                type: 'none',
                values: [
                    'Instrument',
                    'Watercraft',
                ]
            },
            {
                field: 'attribute_facets.keyword',
                type: 'all',
                values: [
                    'Part of Collection',
                    'Access Restriction(s)'
                ]
            }
        ]
    },
    apiConnector: new ElasticSearchConnector({
        host: `${process.env.PUBLIC_URL}/api/search`,
        index: 'demo-index-catalog',
    }),
    alwaysSearchOnInitialLoad: true
};
```

**Expected behavior**

I would expect the 3 global filters to behave as typed out (any/all/none) but from inspecting the connector, this isn't the case. Only the one that overlaps a facet (`kind.keyword`) is included in the final filters passed searchkit, and it's treated as a facet filter rather than the expected global filter behaviour. 

E.g. the constructed query handed off to ES looks like:

```json
POST /demo-index-catalog/_search
{
  "aggs": {
    "facet_bucket_all": {
      "aggs": {
        "category.keyword": {
          "terms": {
            "field": "category.keyword",
            "size": 100,
            "order": {
              "_count": "desc"
            }
          }
        },
        "kind.keyword": {
          "terms": {
            "field": "kind.keyword",
            "size": 100,
            "order": {
              "_count": "desc"
            }
          }
        }
      },
      "filter": {
        "bool": {
          "must": [
            {
              "bool": {
                "must": [
                  {
                    "term": {
                      "kind.keyword": "Instrument"
                    }
                  },
                  {
                    "term": {
                      "kind.keyword": "Watercraft"
                    }
                  }
                ]
              }
            }
          ]
        }
      }
    }
  },
  "size": 20,
  "_source": {
    "includes": [
      "name",
      "description",
      "category",
      "kind"
    ]
  },
  "post_filter": {
    "bool": {
      "must": [
        {
          "bool": {
            "must": [
              {
                "term": {
                  "kind.keyword": "Instrument"
                }
              },
              {
                "term": {
                  "kind.keyword": "Watercraft"
                }
              }
            ]
          }
        }
      ]
    }
  },
  "highlight": {
    "fields": {
      "name": {},
      "description": {},
      "category": {},
      "kind": {}
    }
  },
  "from": 0,
  "sort": [
    {
      "_score": "desc"
    }
  ]
}
```

Looking over the connector code, it does look like global filter type info is ignored in `Request.ts:getFilters` when trying to aggregate these filters and SearchkitRequest.execute's baseFilters argument is not utilized in `index.ts:handleRequest` to pass along anything that would be a non-faceted filter. So I'm not sure if this is by design and my use case is wrong, or if it's still an incomplete feature.

**Which backends and packages are you using:**
Backend: Elasticsearch
Packages: search-ui-elasticsearch-connector

**Comment by joemcelroy** [comment]
thanks for the bug report! Indeed its currently not supported but will add support very soon. Thanks for letting us know.

**Comment by joemcelroy** [comment]
Fix released in v1.13.0. Any issues let me know!
